guides/updating/versions/0.3-to-0.4.md

```diff
@@ -0,0 +1,875 @@
+
+<!--tabs-->
+
+##### 0.3.x
+
+```ts
+import { Agent } from '@aries-framework/core'
+
+// Import from @aries-framework/react-native in React Native
+import { agentDependencies } from '@aries-framework/node'
+
+const agent = new Agent({
+  config: {
+    /* ... */
+    publicDidSeed: '01eafa4de4e22ed4fc2ee522b6ce2731',
+    indyLedgers: [
+      {
+        id: 'Sovrin Mainnet',
+        indyNamespace: 'sovrin',
+        isProduction: true,
+        genesisPath: './genesis/sovrin-genesis.txn',
+      },
+    ],
+  },
+  dependencies: agentDependencies,
+})
+
+await agent.initialize()
+
+const schema = await agent.ledger.registerSchema({
+  attributes: ['name'],
+  name: 'Example Schema',
+  version: '1.0.1',
+})
+
+const credentialDefinition = await agent.ledger.registerCredentialDefinition({
+  schema,
+  supportRevocation: false,
+  tag: 'default',
+})
+```
+
+##### 0.4.x
+
+```ts
+import { AnonCredsModule } from '@aries-framework/anoncreds'
+import { Agent, DidsModule, KeyType, TypedArrayEncoder } from '@aries-framework/core'
+import { IndySdkAnonCredsRegistry, IndySdkModule, IndySdkIndyDidResolver } from '@aries-framework/indy-sdk'
+
+// Import from @aries-framework/react-native in React Native
+import { agentDependencies } from '@aries-framework/node'
+
+// Import from indy-sdk-react-native in React Native
+import indySdk from 'indy-sdk'
+
+const agent = new Agent({
+  config: {
+    /* ... */
+  },
+  dependencies: agentDependencies,
+  modules: {
+    indySdk: new IndySdkModule({
+      indySdk,
+      networks: [
+        {
+          id: 'Sovrin Mainnet',
+          indyNamespace: 'sovrin',
+          isProduction: true,
+          genesisPath: './genesis/sovrin-genesis.txn',
+        },
+      ],
+    }),
+    // Important: register the IndySdkAnonCredsRegistry so that the anoncreds module can resolver/register anoncreds objects based on the networks configured in the IndySdkModule
+    anoncreds: new AnonCredsModule({
+      registries: [new IndySdkAnonCredsRegistry()],
+    }),
+    // Important: register the IndySdkIndyDidResolver so that the anoncreds module can resolve DIDs based on the networks configured in the IndySdkModule
+    dids: new DidsModule({
+      resolvers: [new IndySdkIndyDidResolver()],
+    }),
+  },
+})
+
+// Agent MUST be inialized before the anoncreds module can be used
+await agent.initialize()
+
+// Important: the issuerId (did) MUST be availalbe to the dids module in the agent. You can either create the did using the dids module, or import it as described in this migration guide.
+const schemaResult = await agent.modules.anoncreds.registerSchema({
+  schema: {
+    attrNames: ['name'],
+    issuerId: 'did:indy:sovrin:KEPUoXHcswsfsHkXGfmXB7',
+    name: 'Example Schema',
+    version: '1.0.0',
+  },
+  options: {},
+})
+
+// The resolver and registrar follow the same pattern as the Did Registration
+// and Did Resolver interfaces as used by the did module.
+if (schemaResult.schemaState.state !== 'finished') {
```

Comment:
Might be good, later on, to add a utility method for this like `assertFinished` or `assertState`. 

We would expose both then.

Reply:
Yes that would be a good one to add!